tests/test_enterprise/api_client/test_enterprise.py

```diff
@@ -51,6 +59,54 @@ def _assert_num_requests(self, expected_count):
 
     @responses.activate
     @mock.patch('enterprise.api_client.lms.JwtBuilder', mock.Mock())
+    def test_no_response_doesnt_get_cached(self):
+        """
+        Response doesn't get cached when empty.
+        """
+        uuid = str(self.enterprise_customer.uuid)
+        api_resource_name = 'enterprise-customer'
+        cache_key = get_cache_key(
+            resource=api_resource_name,
+            querystring={},
+            traverse_pagination=False,
+            endpoint_args=(uuid,),
+        )
+
+        cached_enterprise_api_response = cache.get(cache_key)
+        assert cached_enterprise_api_response is None
+
+        self.mock_empty_response('enterprise-customer-courses', uuid)
+        client = enterprise_api.EnterpriseApiClient(self.user)
+        response = client._load_data(  # pylint: disable=protected-access
+            resource=api_resource_name,
+            detail_resource='courses',
+            endpoint_args=(uuid,),
+        )
+        assert not response
+
+        # The empty response is not cached.
+        cached_api_response = cache.get(cache_key)
+        assert not cached_api_response
+
+    @mock.patch('enterprise.api_client.lms.JwtBuilder', mock.Mock())
+    def test_skip_request_if_response_cached(self):
+        """
+        We skip the request portion of the API's logic if the response is already cached.
+        """
+        cache_key = get_cache_key(
+            resource='resource',
+            querystring={},
+            traverse_pagination=False,
+            endpoint_args=(),
+        )
+        cache.set(cache_key, {'fake': 'response'}, settings.ENTERPRISE_API_CACHE_TIMEOUT)
+        client = enterprise_api.EnterpriseApiClient(self.user)
+        response = client._load_data('resource')  # pylint: disable=protected-access
+        cached_api_response = cache.get(cache_key)
```

Comment:
Instead of getting the comparison value out of the cache, let's save the value we cache above into a variable, and then compare against the returned value.

Reply:
:+1: